src/Microsoft.ApiDesignGuidelines.Analyzers/Core/OperatorOverloadsHaveNamedAlternates.Fixer.cs

```diff
@@ -27,10 +32,126 @@ public sealed override FixAllProvider GetFixAllProvider()
+            var generator = context.Document?.Project?.LanguageServices?.GetService<SyntaxGenerator>();
+            if (generator != null)
+            {
+                SyntaxNode addedMember;
+                var bodyStatements = ImmutableArray.Create(
+                    generator.ThrowStatement(generator.ObjectCreationExpression(semanticModel.Compilation.GetTypeByMetadataName("System.NotImplementedException"))));
+                if (OperatorOverloadsHaveNamedAlternatesAnalyzer.IsPropertyExpected(operatorOverloadSymbol.Name))
+                {
+                    // add a property
+                    addedMember = generator.PropertyDeclaration(
+                        name: OperatorOverloadsHaveNamedAlternatesAnalyzer.IsTrueText,
+                        type: generator.TypeExpression(SpecialType.System_Boolean),
+                        accessibility: Accessibility.Public,
+                        modifiers: DeclarationModifiers.ReadOnly,
+                        getAccessorStatements: bodyStatements);
+                }
+                else
+                {
+                    // add a method
+                    var expectedSignature = GetExpectedMethodSignature(operatorOverloadSymbol, semanticModel.Compilation);
+                    addedMember = generator.MethodDeclaration(
+                        name: expectedSignature.Name,
+                        parameters: expectedSignature.Parameters.Select(p => generator.ParameterDeclaration(p.Item1, generator.TypeExpression(p.Item2))),
+                        returnType: generator.TypeExpression(expectedSignature.ReturnType),
+                        accessibility: Accessibility.Public,
+                        modifiers: expectedSignature.IsStatic ? DeclarationModifiers.Static : DeclarationModifiers.None,
+                        statements: bodyStatements);
+                }
+
+                var newTypeDeclaration = generator.AddMembers(typeDeclaration, addedMember);
+                context.RegisterCodeFix(new MyCodeAction(ct => Task.FromResult(context.Document.WithSyntaxRoot(root.ReplaceNode(typeDeclaration, newTypeDeclaration)))),
+                    diagnostic);
+            }
+        }
+
+        private static async Task FixFriendlyVisibility(CodeFixContext context, SyntaxNode node, SemanticModel semanticModel, Diagnostic diagnostic)
+        {
+            var badVisibilityNode = (SyntaxNode)node.FirstAncestorOrSelf<TMethodSyntaxKind>() ?? node.FirstAncestorOrSelf<IPropertySyntaxKind>();
+            var badVisibilitySymbol = semanticModel.GetDeclaredSymbol(badVisibilityNode, context.CancellationToken);
+            var symbolEditor = SymbolEditor.Create(context.Document);
+            var newSymbol = await symbolEditor.EditOneDeclarationAsync(badVisibilitySymbol,
+                (documentEditor, syntaxNode, ct) => Task.Run(() => documentEditor.SetAccessibility(badVisibilityNode, Accessibility.Public))).ConfigureAwait(false);
```

Comment:
Minor: Do you need the Task.Run? I think there's a sync overload for `EditOneDeclarationAsync` as well.